python/az/aro/azext_aro/tests/latest/integration/test_aro_scenario.py

```diff
@@ -2,42 +2,93 @@
 # Licensed under the Apache License 2.0.
 
 import os
+from random import randint
 from unittest import mock
 
-from azure_devtools.scenario_tests import AllowLargeResponse
-from azure.cli.testsdk import ResourceGroupPreparer
-from azure.cli.testsdk import ScenarioTest
 
+from knack.log import get_logger
+from azext_aro.tests.latest.custom_preparers import AROClusterServicePrincipalPreparer
+from azure.cli.testsdk import ScenarioTest, ResourceGroupPreparer
+from azure.cli.testsdk.checkers import StringContainCheck
+from azure.cli.testsdk.scenario_tests import AllowLargeResponse
 
-TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+logger = get_logger(__name__)
 
 
-class AroScenarioTest(ScenarioTest):
-    @ResourceGroupPreparer(name_prefix='cli_test_aro')
-    def test_aro(self, resource_group):
+class AroScenarioTests(ScenarioTest):
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(random_name_length=28, name_prefix='clitestaro_create', location='eastus')
+    @AROClusterServicePrincipalPreparer(name_prefix='clitestaro_create')
+    def test_aro_public_cluster(self, resource_group):
+        from azure.mgmt.core.tools import resource_id
+
+        subscription = self.get_subscription_id()
+
+        master_subnet = self.create_random_name('dev_master', 14)
+        worker_subnet = self.create_random_name('dev_worker', 14)
+        name = self.create_random_name('aro', 14)
+
+        temp_kubeconfig_path = self.create_random_name('kubeconfig', 24) + '.tmp'
+
         self.kwargs.update({
-            'name': 'test1'
+            'name': name,
+            'resource_group': resource_group,
+            'subscription': subscription,
+            'master_subnet': master_subnet,
+            'worker_subnet': worker_subnet,
+            'master_ip_range': '10.{}.{}.0/24'.format(randint(0, 127), randint(0, 255)),
```

Comment:
We don't need these IP ranges to be random if we're making the vnet these are defined on, right?

Reply:
That's true, but the upstream code is written like this, and I can't find any downside to write in this way, so I'll keep it as it is.
https://github.com/Azure/azure-cli/blob/8de4e1305646c0bc931bd50f29377761c5fd324f/src/azure-cli/azure/cli/command_modules/aro/tests/latest/test_aro_commands.py#L34-L44